mongo/results.go

```diff
@@ -13,40 +13,51 @@ import (
 	"go.mongodb.org/mongo-driver/x/mongo/driver/operation"
 )
 
-// BulkWriteResult holds the result of a bulk write operation.
+// BulkWriteResult is the result type returned by a BulkWrite operation.
 type BulkWriteResult struct {
+	// The number of documents inserted.
 	InsertedCount int64
-	MatchedCount  int64
+
+	// The number of documents matched by filters in update and replace operations.
+	MatchedCount int64
+
+	// The number of documents modified by update and replace operations.
 	ModifiedCount int64
-	DeletedCount  int64
+
+	// The number of documents deleted.
+	DeletedCount int64
+
+	// The number of documents upserted by update and replace operations.
 	UpsertedCount int64
-	UpsertedIDs   map[int64]interface{}
+
+	// A map of operation index to the _id of each upserted document.
+	UpsertedIDs map[int64]interface{}
 }
 
-// InsertOneResult is a result of an InsertOne operation.
-//
-// InsertedID will be a Go type that corresponds to a BSON type.
+// InsertOneResult is the result type returned by an InsertOne operation.
 type InsertOneResult struct {
-	// The identifier that was inserted.
+	// The _id of the inserted document. If this was generated by the driver, it will be of type primitive.ObjectID.
```

Comment:
To match what you have below, perhaps just
 
   A value generated by the driver will ...

Reply:
done.